examples/callback/config.yaml

```diff
@@ -0,0 +1,9 @@
+hydra:
+  callbacks:
+    example:
+      _target_: example_callback.ExampleCallback
+    mlflow:
+      _target_: hydra.callback.mlflow.MLflow
+      tracking_uri: file://${hydra:runtime.cwd}/.mlflow
```

Comment:
Experiment with the idea that a callback can be provided by an external package.
This means that the package will be contributing to a config group like hydra/callbacks:

```
hydra/callbacks/
  mlflow.yaml
```

mlflow.yaml
```yaml
# @package _group_
mlflow:
  _target_: hydra.callback.mlflow.MLflow
  tracking_uri: file://${hydra:runtime.cwd}/.mlflow
```

User primary config may now add it:
```yaml
defaults:
  # note that this is not overriding a config group (because they are mutually exclusive for now).
  - hydra/callbacks/mlflow 
  - hydra/callbacks/something_else
```

The above is providing composition order, and hopefully that translates in the resulting config dictionary.